src/System.CommandLine/TypeExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.CommandLine
+{
+    /// <summary>
+    /// Provides extensions to <see cref="Type"/>.
+    /// </summary>
+    public static class TypeExtensions
```

Comment:
These should be kept internal. Maybe include them alongside similar methods in `Binder.cs`?

Reply:
I don't think this is binder specific.

These are extension methods on `Type` that are used by the binder.

It might be a good idea to move all extension methods on `Type` to this class.

Reply:
Within the scope of this project I believe they're only used for binding, which is why e.g. `IsNullable` is also there.

Reply:
I tend to place them on <extended_type>Extensions classes.

If they are super specific, I don't define them as extension methods - maybe one day we can have private extensions.